docs/resources/aws_transit_gateway_attachment.md

```diff
@@ -0,0 +1,102 @@
+---
+page_title: "hcp_aws_transit_gateway_attachment Resource - terraform-provider-hcp"
+subcategory: ""
+description: |-
+  The AWS Transit gateway attachment resource allows you to manage a Transit gateway attachment that attaches an HVN to a Transit gateway in AWS.
+---
+
+# Resource `hcp_aws_transit_gateway_attachment`
+
+The AWS Transit gateway attachment resource allows you to manage a Transit gateway attachment that attaches an HVN to a Transit gateway in AWS.
+
+## Example Usage
+
+```terraform
+provider "aws" {
+  region = "us-west-2"
+}
+
+resource "hcp_hvn" "main" {
+  hvn_id         = "main-hvn"
+  cloud_provider = "aws"
+  region         = "us-west-2"
+  cidr_block     = "172.25.16.0/20"
+}
+
+resource "aws_vpc" "example" {
+  cidr_block = "172.31.0.0/16"
+}
+
+resource "aws_ec2_transit_gateway" "example" {
+  tags = {
+    Name = "example-tgw"
+  }
+}
+
+resource "aws_ram_resource_share" "example" {
+  name                      = "example-resource-share"
+  allow_external_principals = true
+}
+
+resource "aws_ram_principal_association" "example" {
+  resource_share_arn = aws_ram_resource_share.example.arn
+  principal          = hcp_hvn.main.provider_account_id
+}
+
+resource "aws_ram_resource_association" "example" {
+  resource_share_arn = aws_ram_resource_share.example.arn
+  resource_arn       = aws_ec2_transit_gateway.example.arn
+}
+
+resource "hcp_aws_transit_gateway_attachment" "example" {
+  depends_on = [
+    aws_ram_principal_association.example,
+    aws_ram_resource_association.example,
+  ]
+
+  hvn_id                        = hcp_hvn.main.hvn_id
+  transit_gateway_attachment_id = "example-tgw-attachment"
+  transit_gateway_id            = aws_ec2_transit_gateway.example.id
+  resource_share_arn            = aws_ram_resource_share.example.arn
+  destination_cidrs             = [aws_vpc.example.cidr_block]
+}
+
+resource "aws_ec2_transit_gateway_vpc_attachment_accepter" "example" {
+  transit_gateway_attachment_id = hcp_aws_transit_gateway_attachment.example.provider_transit_gateway_attachment_id
+}
+```
+
+## Schema
+
+### Required
+
+- **destination_cidrs** (List of String) The list of associated CIDR ranges.
+- **hvn_id** (String) The ID of the HashiCorp Virtual Network (HVN).
+- **resource_share_arn** (String, Sensitive) The Amazon Resource Name (ARN) of the Resource Share that is needed to grant HCP access to the Transit gateway in AWS.
+- **transit_gateway_attachment_id** (String) The ID of the Transit gateway attachment.
+- **transit_gateway_id** (String) The ID of the Transit gateway in AWS.
+
+### Optional
+
+- **id** (String) The ID of this resource.
+- **timeouts** (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
```

Comment:
1. Do we have any other resources that have nested timeouts? 🤔  
2. Can we add this to the example above?

Reply:
These are actually autogenerated by `go generate`, and allow the user to specify custom timeouts for their operations. We do have these on all of our other resources/data sources currently, so the changes here align with that.

We could include these in the example code, but I think it could be a bit distracting for a user? I would expect a user to only need to set these timeouts in very rare situations, if at all. In the case that these are not useful, we may want to consider just removing them from the documentation (similar to the `id` https://hashicorp.atlassian.net/browse/HCPE-883)

Reply:
Created https://hashicorp.atlassian.net/browse/HCPE-890 to address these `timeouts` fields across all resources